Adding more experimental design with simulated data in ipynb.

sonyahanson opened the pull request:
Added simulated data with varying Kd to ipynb `2b Bayesian fit for two component binding - simulated data.ipynb`.

Commit:
added Kd varying simulated data to ipynb

jchodera commented:
Looks neat!

Some specific oddities I noticed:
* `In[15]` shows that the MCMC sampling is slowly mixing and likely hasn't been run long enough for the automatic equilibration detection to capture the true uncertainty. It would still be great to see if we can figure out which other thermodynamic parameters are moving slowly along with the DeltaG estimate so we can find a way to speed that up.
* `In[46]` has a legend with a very weird ordering: 2 nM, 0.2 nM, 0.02 nM, 20 nM, 200 nM. Maybe those should be in order from smallest to largest? And my understanding was that our feasible range for affinities was more like 1 nM to 1 uM. I imagine that we *could* resolve the 20 pM - 200 nM range if we could get the protein concentration down to 1 nM, but I don't think that's going to produce a measurable signal.

Commit:
updated 2b ipynb legends and sampling plots

sonyahanson commented:
Updated legends, and added plots for other parameter samples in last commit.

Note the point of this ipython notebook is not to show the feasible range of affinities, but rather to show that indeed when the Kd is lower than the protein concentration our resolving power for delG estimates gets worse. 

I realize this image gets cutoff in the github rendering of the notebook, and I'm not sure how to fix that right now, but here is the resulting figure that shows this (already posted on Slack a while ago):
![image](https://cloud.githubusercontent.com/assets/6333823/23719763/9bca8496-040a-11e7-8455-f5937af7ba9d.png)
[Vertical lines are real Kd, histogram is our prediction.]

sonyahanson commented:
I'm going to merge this, as I'd like to try and implement the `emcee` sampling methods to see if this improves the equilibration of the trace for the delG, since now it doesn't look that great. Though the result itself is spot on:

![image2](https://cloud.githubusercontent.com/assets/6333823/23720099/b443b924-040b-11e7-859e-2ac49f28c85b.png)
[Only blue points (after eqil) used in histogram, real delG is -20.03 k_B T.]
